File: workspace/java/src/test/java/com/dslplatform/ocd/values/TestArrayLongValue.java
package com.dslplatform.ocd.values;

import com.dslplatform.client.Bootstrap;
import com.dslplatform.ocd.values.ArrayLongInValue.ArrayLongValue;
import com.dslplatform.patterns.ServiceLocator;
import org.junit.*;
import static org.junit.Assert.*;

public class TestArrayLongValue {
    private static ServiceLocator locator;

    @BeforeClass
    public static void setUpClass() throws Exception {
//        locator = Bootstrap.init(TestArrayLongValue.class.getResourceAsStream("dsl-project.ini"));
    }

    @AfterClass
    public static void tearDownClass() throws Exception {
//        locator.resolve(java.util.concurrent.ExecutorService.class).shutdown();
//        locator = null;
    }

    @Before
    public void setUp() throws Exception {
    }

    @After
    public void tearDown() throws Exception {
    }

    @Test
    public void testFieldType() throws NoSuchFieldException {
        assertEquals(
                long[].class,
                ArrayLongValue.class.getDeclaredField("arrayLong").getGenericType());
    }

    @Test
    public void testGetterType() throws NoSuchMethodException {
        assertEquals(
                long[].class,
                ArrayLongValue.class.getMethod("getArrayLong").getGenericReturnType());
    }

    @Test
    public void testSetterType() throws NoSuchMethodException {
        assertEquals(
                ArrayLongValue.class,
                ArrayLongValue.class.getMethod("setArrayLong", long[].class).getReturnType());
    }

    @Test
    public void testDefaultPropertyValue() {
        assertArrayEquals(
                new long[0],
                new ArrayLongValue().getArrayLong());
    }
}

File: workspace/scala/src/test/scala/com/dslplatform/ocd/values/TestArrayLongValue.scala
package com.dslplatform.ocd.values

import ArrayLongInValue.ArrayLongValue
import com.dslplatform.api.patterns.ServiceLocator
import org.scalatest._
import scala.reflect.runtime.universe._

class TestArrayLongValue extends SpecExtensions {
  def `Test property field type: Array[Long]`() =
    checkType(
      typeOf[Array[Long]]
    , typeOf[ArrayLongValue].member("arrayLong": TermName).asMethod.returnType
    )

  def `Test property default value: Array.empty[Long]`() {
    assert(ArrayLongValue().arrayLong === Array.empty[Long])
  }
}
